src/toil/jobStores/aws/jobStore.py

```diff
@@ -349,28 +389,54 @@ def _getKeyForUrl(url, existing=None):
                exists and it will be created. If None, the key will be created if it doesn't exist.
 
         :rtype: Key
+        >>> import urlparse
```

Comment:
I was trying to create the below code so that it can be covered in a test. I think I might have misunderstood what you had in mind for this test.